mprime-python-port/exp.py

```diff
@@ -20,28 +20,25 @@
 
 # Prerequisites, gained from mprime.py
 USERID, COMPUTER, TYPE  = sys.argv[1], sys.argv[2], sys.argv[3]
+PROOF_CERTIFICATION_WORK = "n"
 
 child = pexpect.spawn('./mprime -m') # starts shell to interact with
 child.logfile = sys.stdout.buffer # enables output to screen (Python 3)
-expects = (("Join Gimps?", "y"), ("Use PrimeNet to get work and report results ()", "y"),
-        ("Your user ID or", USERID), ("Optional computer name", COMPUTER),
-        ("Type of work to get", TYPE), ("Your choice:", "5"))
-index = 0
+expects = ["Join Gimps?", "Use PrimeNet to get work and report results ()", "Your user ID or", 
+           "Optional computer name", "Type of work to get", "CPU cores to use", 
+           "Your choice:", pexpect.TIMEOUT, "Done communicating with server.",
+           "Choose Test/Continue to restart","Upload bandwidth limit in Mbps", "Skip advanced resource settings",
+           "stage 2 memory in GB", "Max emergency memory in GB/worker", "Get occasional proof certification work"]
+responses = ["y", "y", USERID, 
+             COMPUTER, TYPE, "", 
+             "5", "", "\x03", 
+             "5", "10000", "n", 
+             "12", "3", "n"]
```

Comment:
I do not think we want to put the Intel specific stuff in this script, otherwise it will only work on that system. I will need to create a new `sed` command for the notebooks to dynamically add it in, as the both your Colab notebooks currently do for my regular Expect script.

In addition, maybe put these two lists in a dictionary so it is easier to read and maintain. That should also make it easier for the `sed` command to modify. Then convert it back to lists with something like this:
```py
expects = list(dict.keys())
responses = list(dict.values())
```
The “CPU cores to use” is also not used here.

Reply:
Can you specify which is intel specific? It may have been a while since I got to this and can't remember.

Really nice idea about the dictionary, I was annoyed that pexpect does not support dictionaries, but that is a super readable alternative!

Reply:
> Can you specify which is intel specific?

All the items you added to the lists.

No problem.